src/Microsoft.Diagnostics.Runtime/src/DacInterface/ClrDataModule.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Microsoft.Diagnostics.Runtime.Utilities;
+
+namespace Microsoft.Diagnostics.Runtime.DacInterface
+{
+    public sealed unsafe class ClrDataModule : CallableCOMWrapper
+    {
+        private const uint DACDATAMODULEPRIV_REQUEST_GET_MODULEDATA = 0xf0000001;
+
+        private static readonly Guid IID_IXCLRDataModule = new Guid("88E32849-0A0A-4cb0-9022-7CD2E9E139E2");
+
+        private RequestDelegate? _request;
+
+        public ClrDataModule(DacLibrary library, IntPtr pUnknown)
+            : base(library?.OwningLibrary, IID_IXCLRDataModule, pUnknown)
+        {
+        }
+
+        private ref readonly IClrDataModuleVTable VTable => ref Unsafe.AsRef<IClrDataModuleVTable>(_vtable);
+
+        internal bool GetModuleData(out UntitledModuleData data)
```

Comment:
You should mark this (and the struct) public.  We do expose these things publicly in case someone is using ClrMD to build their own implementation.